content/blog/getting-started-with-java-development-2023.asciidoc

```diff
@@ -0,0 +1,211 @@
+---
+title: "Getting Started With Java Development in 2023 -- An Opinionated Guide"
+date: 2023-01-15T22:10:00+01:00
+draft: false
+markup: adoc
+---
+:source-highlighter: rouge
+:rouge-style: base16.dark
+:icons: font
+:imagesdir: /images
+ifdef::env-github[]
+:imagesdir: ../../static/images
+endif::[]
+
+27 years of age, and alive and kicking -- The Java platform regularly comes out amonst the top contenders in rankings like the https://www.tiobe.com/tiobe-index/[TIOBE index].
+In my opinion, rightly so. The language is very actively maintained and constantly improved;
+its underlying runtime, the Java Virtual Machine (JVM),
+is one of, if not the most, advanced runtime environments for managed programming languages.
+There is a massive eco-system of Java libraries which make it a great tool for a large number of use cases,
+ranging from command-line and desktop applications over, web apps and backend web services, to datastores and stream processing platforms.
+With upcoming features like support for vectorized computations (SIMD),
+improved integration with native code (Project Panama),
+value objects and user-defined primitives (Project Valhalla), and others,
+Java is becoming a great tool for solving a larger number of software development tasks than ever before.
+
+<!--more-->
+
+The immense breadth of Java and its ecosystem, having grown and matured for nearly three decades,
+can it also make a bit tough though to get started for folks just starting their careers as a Java developer.
+Which version should you use? How to install it? Which build tool and IDE are the right ones?
+For all these, and many other questions, there are typically a number of options,
+which can easily overwhelm you if you are new to Java.
+As the platform has evolved, tools have come and gone,
+things which were hugely popular years ago have fallen into obsolecense since then.
+Related information can be found on the internet,
+making it challenging to identify what's still relevant and what not.
+
+The idea for this blog post is to provide an opinionated guide for folks getting started with Java development in 2023,
+helping you with your very first steps with that amazing platform.
+Note I'm not saying that the things I'm going to recommend are the best ones for each and every situation.
+The focus is on providing a good getting-started experience,
+some of the recommended tools or approaches may make less sense to use as you get more experienced.
+Other choices might be better suited for you later on, based on the specific situation and its requirements.
+Also there is a notion of personal taste and preference to these things,
+those are _my_ recommendations, and those of others might look different, which is perfectly fine.
+
```

Comment:
Imho it could be helpful to explain ( in short) "Java (language)", "JDK", "JRE", "JVM".

Also might be worth mentioning that there are other languages for the java platform (like groovy, kotlin)